### PR TITLE
[stable/mariadb] quote annotation value

### PR DESCRIPTION
**Pull request by foosinn**
#### What this PR does / why we need it:

Currently annotations with special characters are broken. Example:

```yaml
      annotations:
        backup.velero.io/backup-volumes: data
        pre.hook.backup.velero.io/command: ["sh", "-c", "/opt/bitnami/mariadb/bin/mysqldump -A -u root -p$MARIADB_ROOT_PASSWORD | /bin/gzip > /bitnami/mariadb/full_dump.gz"]

```

This PR quotes the annotantions value.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [X] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [X] Chart Version bumped
- [X] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

File: stable/mariadb/templates/master-statefulset.yaml [hunk 1]
```diff
@@ -26,7 +26,7 @@ spec:
       {{- if .Values.master.annotations }}
       annotations:
         {{- range $key, $value := .Values.master.annotations }}
-        {{ $key }}: {{ $value }}
+        {{ $key }}: {{ $value | quote }}
```
Comment (javsalgar) [inline stable/mariadb/templates/master-statefulset.yaml]: if $value is an array (just like the example you mentioned) what would the output look like?

**Review by javsalgar — COMMENTED** [review]
Thanks for the PR. Small question

File: stable/mariadb/templates/master-statefulset.yaml (hunk 1 above)
Comment (foosinn) [reply stable/mariadb/templates/master-statefulset.yaml]: sorry, i might have not been clear.

currently we have this in our values.yaml

```yaml
mariadb:
  master:
    annotations:
      pre.hook.backup.velero.io/command: '["sh", "-c", "/opt/bitnami/mariadb/bin/mysqldump -A -u root -p$MARIADB_ROOT_PASSWORD | /bin/gzip > /bitnami/mariadb/full_dump.gz"]'
```

this results to:
```yaml
apiVersion: apps/v1
kind: StatefulSet
metadata:
  name: ***
  labels:
    app: mariadb
    chart: mariadb-7.3.4
    release: ***
    heritage: Helm
    component: master
spec:
  selector:
    matchLabels:
      app: mariadb
      release: ***
      component: master
  serviceName: ****
  replicas: 1
  updateStrategy:
    type: RollingUpdate
  template:
    metadata:
      annotations:
        backup.velero.io/backup-volumes: data
        pre.hook.backup.velero.io/command: ["sh", "-c", "/opt/bitnami/mariadb/bin/mysqldump -A -u root -p$MARIADB_ROOT_PASSWORD | /bin/gzip > /bitnami/mariadb/full_dump.gz"]
```

so the string gets convertet to a yaml list but not quoting it.

if i put in a yaml list (not shure why i should to that) i get the go %v representation of a list:

```yaml
        pre.hook.backup.velero.io/command: [sh -c /opt/bitnami/mariadb/bin/mysqldump -A -u root -p$MARIADB_ROOT_PASSWORD | /bin/gzip > /bitnami/mariadb/full_dump.gz]
```

Comment (foosinn) [reply stable/mariadb/templates/master-statefulset.yaml]: short update, i noticed while working with some other charts that they seem to use toYaml nowadays.

PR is updated  with and toYaml syntax.

**Comment by foosinn** [comment]
Hey, are there any changes left to be done?

**Pull request merged**